yjit/src/codegen.rs

```diff
@@ -68,10 +68,25 @@ impl JITState {
         }
     }
 
+    pub fn new_with_opcode(opcode: usize) -> Self {
+        JITState {
+            block: Block::new(BLOCKID_NULL),
+            iseq: IseqPtr(0),
+            insn_idx: 0,
+            opcode,
+            side_exit_for_pc: CodePtr::null(),
+            record_boundary_patch_point: false,
+        }
+    }
+
     pub fn add_gc_object_offset(self:&mut JITState, ptr_offset:u32) {
         let mut gc_obj_vec: RefMut<_> = self.block.borrow_mut();
         gc_obj_vec.add_gc_object_offset(ptr_offset);
     }
+
+    pub fn get_opcode(self:&JITState) -> usize {
+        self.opcode
+    }
```

Comment:
Going to show how new to Rust I am here but do we need an accessor method to access the opcode within the same module?